### DISCUSSION TITLE
Cannot open code in visual studio

### DISCUSSION THREAD

**Discussion by Rett1022** [community/community#67213] [GitHub Education]
### Hi everyone,

I am trying to open code from a link for a class and I keep getting this error message:

Open Assignment: Failed to run 'git clone'. fatal: Too many arguments.

usage: git clone [] [--] [

]
-v, --verbose         be more verbose
-q, --quiet           be more quiet
--progress            force progress reporting
--reject-shallow      don't clone shallow repository
-n, --no-checkout     don't create a checkout
--bare                create a bare repository
--mirror              create a mirror repository (implies bare)
-l, --local           to clone from a local repository
--no-hardlinks        don't use local hardlinks, always copy
-s, --shared          setup as shared repository
--recurse-submodules[=<pathspec>]
                      initialize submodules in the clone
--recursive ...       alias of --recurse-submodules
-j, --jobs <n>        number of submodules cloned in parallel
--template <template-directory>
                      directory from which templates will be used
--reference <repo>    reference repository
--reference-if-able <repo>
                      reference repository
--dissociate          use --reference only while cloning
-o, --origin <name>   use <name> instead of 'origin' to track upstream
-b, --branch <branch> checkout <branch> instead of the remote's HEAD
-u, --upload-pack <path>
                      path to git-upload-pack on the remote
--depth <depth>       create a shallow clone of that depth
--shallow-since <time>
                      create a shallow clone since a specific time
--shallow-exclude <revision>
                      deepen history of shallow clone, excluding rev
--single-branch       clone only one branch, HEAD or --branch
--no-tags             don't clone any tags, and make later fetches not to follow them
--shallow-submodules  any cloned submodules will be shallow
--separate-git-dir <gitdir>
                      separate git dir from working tree
-c, --config <key=value>
                      set config inside the new repository
--server-option <server-specific>
                      option to transmit
-4, --ipv4            use IPv4 addresses only
-6, --ipv6            use IPv6 addresses only
--filter <args>       object filtering
--also-filter-submodules
                      apply partial clone filters to submodules
--remote-submodules   any cloned submodules will use their remote-tracking branch
--sparse              initialize sparse-checkout file to include only files at root
--bundle-uri <uri>    a URI for downloading bundles before fetching from origin remote
I'm using the github classroom extension.

**Comment by elendil7** [comment]
### Step 1

Open a command prompt window (CMD). This can be done by pressing the Windows key on your keyboard (or clicking on the search bar at the bottom left of the screen), and typing "cmd", then clicking on the popup application.

### Step 2

Make sure you're running the command properly, to clone a github repository to your local disk:

```bash
git clone https://github.com/<username>/<reponame>
```

Example:
```bash
git clone https://github.com/Rett1022/exampleRepository
```

### Step 3

Then, to open the cloned git repository in VSCode:

- CD into the directory that was generated by git clone. E.g., ```cd exampleRepository```
- Run ```cd``` or ```chdir```, and copy the directory path to your clipboard using the CTRL+C shortcut.
- Navigate to the Visual Studio Code (VSCode) application.
- Click F1, or hold the CTRL+SHIFT+P keys to open a search bar at the top of the screen.
- Type in "open folder", and you should see one option which matches the text you gave. Hit the enter key.
- Now, you should see a file explorer window. Hit the CTRL+L shortcut (or left click on the searchbar at the top of the file explorer window), and paste the text you copied earlier using the CTRL+V shortcut.
- Now, click enter. Then, at the bottom right hand side of the file explorer window, click OK.
- VSCode should now open your code project!

#### Note

If you're using Visual Studio instead of Visual Studio Code, you can follow the start of the guide linked below, to open the folder:

https://learn.microsoft.com/en-us/visualstudio/ide/develop-code-in-visual-studio-without-projects-or-solutions?view=vs-2022
